### PR TITLE
[WIP] Number precision changes

### PR DESCRIPTION
**Pull request by amangale**
This PR adds the number precision changes to openbazaar-go through CurrencyValue which uses big.Int.

### CODE REVIEW

File: api/jsonapi_test.go [hunk 1]
```diff
@@ -440,7 +444,7 @@ func TestCryptoListingsIllegalFields(t *testing.T) {
 	physicalListing := factory.NewListing("physical")
 
 	listing := factory.NewCryptoListing("crypto")
-	listing.Metadata.PricingCurrency = "btc"
+	listing.Metadata.PricingCurrency = &pb.CurrencyDefinition{Code: "BTC", Divisibility: 8}
```
Comment (placer14) [inline api/jsonapi_test.go]: The preferred way to create CurrencyDefinitions is with `repo.LoadCurrencyDefinitions().Lookup("BTC")`.

File: api/jsonapi_test.go [hunk 2]
```diff
@@ -458,12 +462,13 @@ func TestCryptoListingsIllegalFields(t *testing.T) {
...
 	listing.Coupons = physicalListing.Coupons
 	runTest(listing, core.ErrCryptocurrencyListingIllegalField("coupons"))
+
 }
 
 func TestMarketRatePrice(t *testing.T) {
 	listing := factory.NewListing("listing")
 	listing.Metadata.Format = pb.Listing_Metadata_MARKET_PRICE
-	listing.Item.Price = 1
+	listing.Item.Price = &pb.CurrencyValue{Currency: &pb.CurrencyDefinition{Code: "BTC", Divisibility: 8}, Value: "1"}
```
Comment (placer14) [inline api/jsonapi_test.go]: You can use `repo.NewCurrencyValue("1", btcDef)` with the `CurrencyDefinition` you looked up above to properly generate valid objects.

File: api/jsonapi_test.go [hunk 3]
```diff
@@ -570,7 +575,7 @@ func TestCloseDisputeBlocksWhenExpired(t *testing.T) {
 func TestZECSalesCannotReleaseEscrow(t *testing.T) {
 	sale := factory.NewSaleRecord()
 	sale.Contract.VendorListings[0].Metadata.AcceptedCurrencies = []string{"ZEC"}
-	sale.Contract.BuyerOrder.Payment.Coin = "ZEC"
+	sale.Contract.BuyerOrder.Payment.Amount = &pb.CurrencyValue{Currency: &pb.CurrencyDefinition{Code: "ZEC", Divisibility: 8}}
```
Comment (placer14) [inline api/jsonapi_test.go]: Please use `repo.NewCurrencyValue`.

**Comment by cpacia** [comment]
Closing to re-open with a branch inside ob-go

**Pull request closed without merging**